PhpAmqpLib/Wire/AMQPAbstractCollection.php

```diff
@@ -467,26 +471,31 @@ final public static function getDataTypeForSymbol($symbol)
         return $symbols[$symbol];
     }
 
+    #[\ReturnTypeWillChange]
     public function current()
```

Comment:
Failed test: https://github.com/javer/php-amqplib/runs/3210150874?check_suite_focus=true#step:11:13
```
PHP Fatal error:  During inheritance of Iterator: Uncaught Return type of PhpAmqpLib\Wire\AMQPAbstractCollection::current() should either be compatible with Iterator::current(): mixed, or the #[ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPAbstractCollection.php:470
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPAbstractCollection.php:12
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php:365
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPWriter.php:383
/home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Message/AMQPMessage.php:529
/home/runner/work/php-amqplib/php-amqplib/tests/Unit/Message/AMQPMessageTest.php:21
 in /home/runner/work/php-amqplib/php-amqplib/PhpAmqpLib/Wire/AMQPAbstractCollection.php on line 12
```

Since PHP 8.1 many internal classes have got correct return types: https://wiki.php.net/rfc/internal_method_return_types
And we cannot specify all of them explicitly, because this package supports PHP 7.0 which doesn't support all necessary return types. So as a workaround till PHP 9.0 we can use `ReturnTypeWillChange` attribute which suppresses this error.

The same reason for another the same changes.

Reply:
What is minimum required versions to make definitions compatible?

Reply:
As we can see from the error message `should either be compatible with Iterator::current(): mixed` it requires `mixed` type which was introduced in PHP 8.0.